src/httpd_jsonapi.c

```diff
@@ -3176,6 +3176,72 @@ jsonapi_reply_library_tracks_put_byid(struct httpd_request *hreq)
   return HTTP_OK;
 }
 
+static int
+jsonapi_reply_library_track_playlists(struct httpd_request *hreq)
+{
+  struct query_params query_params;
+  json_object *reply;
+  json_object *items;
+  char *path;
+  const char *track_id;
+  int id;
+  int total;
+  int ret = 0;
+
+  if (!is_modified(hreq->req, DB_ADMIN_DB_MODIFIED))
+    return HTTP_NOTMODIFIED;
+
+  track_id = hreq->uri_parsed->path_parts[3];
+  if (safe_atoi32(track_id, &id) < 0)
+    {
+      DPRINTF(E_LOG, L_WEB, "Error converting track id '%s' to int.\n", track_id);
+      return HTTP_INTERNAL;
+    }
+
+  path = db_file_path_byid(id);
+  if (!path)
+    {
+      DPRINTF(E_WARN, L_WEB, "No file path found for track with id '%s' not found.\n", track_id);
+      return HTTP_BADREQUEST;
+    }
+
+  reply = json_object_new_object();
+  items = json_object_new_array();
+  json_object_object_add(reply, "items", items);
+
+  memset(&query_params, 0, sizeof(struct query_params));
+
+  ret = query_params_limit_set(&query_params, hreq);
+  if (ret < 0)
+    goto error;
+
+  query_params.type = Q_PL;
```

Comment:
Maybe you could use Q_FIND_PL instead? It might make the filter more straightforward. You can see an example of how it is used in source_item_ownpl_get() in artwork.c.